pkg/event/target/memorystore.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package target
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/minio/minio/pkg/event"
+)
+
+const (
+	maxStoreLimit = 10000
+)
+
+// MemoryStore persists events in memory.
+type MemoryStore struct {
+	sync.RWMutex
+	events map[string]event.Event
+	eC     uint16
+	limit  uint16
+}
+
+// NewMemoryStore creates a memory store instance.
+func NewMemoryStore(limit uint16) *MemoryStore {
+	if limit == 0 || limit > maxStoreLimit {
+		limit = maxStoreLimit
+	}
+	memoryStore := &MemoryStore{
+		events: make(map[string]event.Event),
+		limit:  limit,
+	}
+	return memoryStore
+}
+
+// Open is in-effective here.
+// Implemented for interface compatibility.
+func (store *MemoryStore) Open() error {
+	return nil
+}
+
+// Put - puts the event in store.
+func (store *MemoryStore) Put(e event.Event) {
```

Comment:
Put() should return an error upon limit with ErrLimitExceeded

Reply:
Done